### PR TITLE
feat: duplicate init split prebuilt UI

### PR DESCRIPTION
**Pull request by alisher-aituarov**
## Summary of change

Combination of https://github.com/supertokens/supertokens-auth-react/pull/640 & https://github.com/supertokens/supertokens-auth-react/pull/655

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.
-   [ ] If I added a new recipe, I also added the recipe entry point into the `rollup.config.mjs`

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: examples/for-tests/src/AppWithoutRouter.js [hunk 1]
```diff
@@ -12,9 +17,38 @@ function AppWithoutRouter() {
...
+
+let recipePreBuiltUI = EmailPasswordPreBuiltUI;
+if (authRecipe === "thirdparty") {
+    recipePreBuiltUI = ThirdPartyPreBuiltUI;
+} else if (authRecipe === "emailpassword") {
+    recipePreBuiltUI = EmailPasswordPreBuiltUI;
+} else if (authRecipe === "both") {
+    recipePreBuiltUI = {
+        canHandleRoute: () => {
+            return ThirdPartyPreBuiltUI.canHandleRoute() && EmailPasswordPreBuiltUI.canHandleRoute();
```
Comment (porcellus) [inline examples/for-tests/src/AppWithoutRouter.js:30]:
```suggestion
            return ThirdPartyPreBuiltUI.canHandleRoute() || EmailPasswordPreBuiltUI.canHandleRoute();
```

**Review by porcellus — CHANGES_REQUESTED** [review]
Please add a check to examples you manual tested (and test the missing ones) using the latest version:
- [x] with-next-iframe
- [x] with-multiple-email-sign-in
- [x] with-phone-password
- [ ] with-sign-in-up-split-emailpassword
- [ ] with-svelte-react-thirdpartyemailpassword
- [ ] with-thirdpartyemailpassword-2fa-passwordless
- [ ] with-thirdpartyemailpassword-passwordless
- [ ] with-thirdpartypasswordless-electron

Plus test that the new branch of create-supertokens-app creates a working example for different configs:
- [ ] angular & node w/ thirdpartyemailpassword
- [ ] react & node w/ thirdpartyemailpassword
- [ ] vue & node w/ thirdpartyemailpassword
- [ ] nextjs
- [ ] capacitor

**Review by porcellus — CHANGES_REQUESTED** [review]
Please complete the above checklist after the latest changes (I've unchecked everything)

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,123 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:8]: change in this

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,8 +7,123 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+### Changes
+
+-   Eliminated the need for duplicate `init` call for non-react applications that use pre-built UI. [See the issue](https://github.com/supertokens/supertokens-auth-react/issues/616)
+-   Split/separate pre-built UI components from its recipe to reduce bundle sizes for apps that do not use pre-built UI
 -   Added thirdparty login with popup window example
 
+### Migration
+
```
Comment (rishabhpoddar) [inline CHANGELOG.md:17]: add that we should not init both web-js and auth-react -> just one of them

File: examples/for-tests-react-16/src/AppWithReactDomRouter.js [hunk 4]
```diff
@@ -1,14 +1,42 @@
 import React, { useState } from "react";
```
Comment (rishabhpoddar) [inline examples/for-tests-react-16/src/AppWithReactDomRouter.js:1]: need to update google one tap example app (if this does not work on chrome, try firefox)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**